### PR TITLE
use pytest for integration tests

### PR DESCRIPTION
**Pull request by johnandersen777**
Per https://github.com/DavidBuchanan314/millipds/issues/16 TODO use testing framework (pytest)

### CODE REVIEW

File: tests/integration_test.py [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+import aiohttp
+
+from millipds import service
+from millipds import database
+from millipds import crypto
+
+@dataclasses.dataclass
+class TestPDS:
+	endpoint: str
+	db: database.Database
```
Comment (johnandersen777) [inline tests/integration_test.py:17]: I did this because future tests may need the database

**Comment by DavidBuchanan314** [comment]
Thank you for the contribution, this is looking really clean!

File: tests/integration_test.py [hunk 2]
```diff
@@ -10,140 +38,231 @@
...
 	TEST_HANDLE = "local.dev.retr0.id"
 	TEST_PASSWORD = "lol"
+TEST_PRIVKEY = crypto.keygen_p256()
+
+@pytest.fixture
+async def test_pds(aiolib):
+	queue = asyncio.Queue()
+	with tempfile.TemporaryDirectory() as tempdir:
+		db_path = f"{tempdir}/millipds-0000.db"
+		db = database.Database(path=db_path)
```
Comment (DavidBuchanan314) [inline tests/integration_test.py:48]: I believe passing `:memory:` will make sqlite use an in-memory db, which would avoid the need for a temp path. But, I suppose tests could plausibly involve writing larger-than-memory amounts of data in the future, so let's keep it.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -1,6 +1,12 @@
 name: Run tests
 run-name: ${{ github.actor }} is running tests
-on: push
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - main
```
Comment (DavidBuchanan314) [inline .github/workflows/test.yml:9]: I need to do a little more research to make sure I understand the security implications of running CI against PRs

Comment (DavidBuchanan314) [reply .github/workflows/test.yml:9]: The github actions security model seems a little convoluted and I'm not sure I fully understand it, but I went into the settings and set it to "always require approval" to run on PRs, which I think should keep things pretty safe.

Comment (johnandersen777) [reply .github/workflows/test.yml:9]: Requires approval is always safe, the threat model is secure by default. Things that break the threat model are basically `pull_request_target` or anything that allows use of secrets from PRs or write access to repo contents via PRs. [The whole PyPi release flow can be done secretless these days via OIDC](https://docs.pypi.org/trusted-publishers/using-a-publisher/#github-actions).

https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/

**Pull request merged**